### PR TITLE
Parse JSON into org.embulk.spi.json.JsonValue, not org.msgpack.value.Value

### PR DESCRIPTION
**Pull request by dmikurube**
As commented in `embulk-util-json`, we are going to release `embulk-parser-json:0.11.1` just with dropping use of `org.msgpack.value.Value`, but without one-path parsing.

* https://github.com/embulk/embulk-util-json/pull/33#issuecomment-1727651903

This pull request includes :
* Replacing use of the old `JsonParser` to new `JsonValueParser`, with replacing use of `org.msgpack.value.Value` to `org.embulk.spi.json.JsonValue`
* Testing it with `org.embulk.spi.json.JsonValue`, without `org.msgpack.value.Value`
* Removing unnecessary dependencies

Each step is tested as an independent commit on CI.

`embulk-parser-json:0.11.1` would be released with this, and also Embulk v0.11.1.

### CODE REVIEW

File: src/main/java/org/embulk/parser/json/JsonParserPlugin.java [hunk 1]
```diff
@@ -167,40 +163,54 @@ public void run(TaskSource taskSource, Schema schema, FileInput input, PageOutpu
...
                             }
 
-                            final Iterable<Value> recordValues;
+                            final Iterable<JsonValue> recordValues;
                             if (task.getFlattenJsonArray()) {
-                                if (!value.isArrayValue()) {
+                                if (!value.isJsonArray()) {
                                     throw new JsonRecordValidateException(
                                             String.format(
                                                     "A Json record must represent array value with '__experimental__flatten_json_array' option, but it's %s",
```
Comment (hiroyuki-sato) [inline src/main/java/org/embulk/parser/json/JsonParserPlugin.java]: Offtopic: `__experimental__flatten_json_array` -> `flatten_json_array`

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍 (Please check the one comment)

File: src/main/java/org/embulk/parser/json/JsonParserPlugin.java (hunk 1 above)
Comment (dmikurube) [reply src/main/java/org/embulk/parser/json/JsonParserPlugin.java]: Thanks for your good catch! Exactly. Fixed it in: b8039b48869f43c58c1a6a3c9fab947751c76252

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**